lab_1_classify_by_unigrams/main.py

```diff
@@ -19,6 +30,16 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list) or not all(isinstance(token, str) for token in tokens):
+        return None
+
+    dict_of_freq = {}
+    for token in tokens:
+        quantity_of_token = tokens.count(token)
+        frequency = quantity_of_token / len(tokens)
```

Comment:
I believe the world be :
1. Do not calculate `len(tokens)` every iteration
2. Do not calculate the save frequencies for the same tokens many times